src/lib.rs

```diff
@@ -442,6 +505,7 @@ where
 {
     decorator: D,
     fn_timestamp: Box<dyn ThreadSafeTimestampFn<Output = io::Result<()>>>,
+    header_printer: Option<Box<dyn ThreadSafeHeaderFn>>,
```

Comment:
Do we really need this Option? Couldn't we just start with default, and allow overwriting it, just like `fn_timestamp` already is?